### ISSUE TITLE
Weird base dev-server path with esbuild 

### ISSUE THREAD

**Issue by paoyuan** [angular/angular-cli#25195]
### Command

serve

### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### Description

Following #25128 the base dev-server path has been set (ex. admin)
1. Open http://localhost:4200, the page url has two base path like http://localhost:4200/admin/#/admin/dashboard
2. Open asset file http://localhost:4200/admin/assets/css/bootstrap.min.css, it shows
`The server is configured with a public base URL of /admin/ - did you mean to visit /admin/@fs/D:\MyProject\admin\src\assets\css\bootstrap.min.css instead?`
You have to change it to http://localhost:4200/admin/@fs/D:/MyProject/admin/src/assets/css/bootstrap.min.css to make it work properly

### Minimal Reproduction

npm run start
ng serve --hmr

### Exception or Error

```text
Failed to load url /admin/assets/css/bootstrap.min.css (resolved id: /admin/assets/css/bootstrap.min.css). Does the file exist?
```

### Your Environment

```text
Angular CLI: 16.0.1
Node: 18.16.0
Package Manager: npm 7.17.0
OS: win32 x64

Angular: 16.0.1
... animations, cdk, cli, common, compiler, compiler-cli, core 
... forms, material, platform-browser, platform-browser-dynamic
... router

Package                         Version
---------------------------------------------------------      
@angular-devkit/architect       0.1600.1
@angular-devkit/build-angular   16.0.1
@angular-devkit/core            16.0.1
@angular-devkit/schematics      16.0.1
@schematics/angular             16.0.1
rxjs                            6.6.7
typescript                      5.0.4
```

**Comment by paoyuan** [comment]
@clydin With new version 16.0.2, this issue is still present.
1. Web page is working with double base path
2. Same error msg
`The server is configured with a public base URL of /admin/ - did you mean to visit /admin/@fs/D:\MyProject\admin\src\assets\css\bootstrap.min.css instead?`

**Comment by sumitparakh** [comment]
Hi @alan-agius4, can I take this up?

**Comment by paoyuan** [comment]
Hi @clydin,
I found that this issue occurred with location strategy useHash=true.

```
/**
     * When true, enable the location strategy that uses the URL fragment
     * instead of the history API.
  */
useHash?: boolean;
```

**Comment by paoyuan** [comment]
This issue is solved with the latest version of 16.2.0.
